src/api_helpers.jl

```diff
@@ -37,10 +37,40 @@ function h5a_get_name_by_idx(loc_id, obj_name, idx_type, order, idx, lapl_id)
     return String(buf)
 end
 
-function h5a_iterate(f, obj_id, idx_type, order, idx = 0)
+# Some architectures do not support runtime closures with @cfunction, so we instead
+# emulate the behavior by passing the intended callback (which may be a closure) through
```

Comment:
```suggestion
# hdf supports proper closure environments, so we use that support rather than
# emulating it with the less desirable form of creating closure handles directly in cfunction with $f.
# This helper translates between the two preferred forms for each respective language.
```